include/boost/geometry/io/svg/svg_mapper.hpp

```diff
@@ -292,37 +292,85 @@ class svg_mapper : boost::noncopyable
+
+            auto const& m = std::max(w, h) * (1.0 - m_scale);
+            geometry::set<0, 0>(b, geometry::get<0, 0>(b) - m);
+            geometry::set<0, 1>(b, geometry::get<0, 1>(b) - m);
+            geometry::set<1, 0>(b, geometry::get<1, 0>(b) + m);
+            geometry::set<1, 1>(b, geometry::get<1, 1>(b) + m);
+        }
+    }
 
     void init_matrix()
     {
         if (! m_matrix)
         {
+            scale_bounding_box();
             m_matrix.reset(new transformer_type(m_bounding_box,
                             m_width, m_height));
-
-
-            m_stream << "<?xml version=\"1.0\" standalone=\"no\"?>"
-                << std::endl
-                << "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\""
-                << std::endl
-                << "\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">"
-                << std::endl
-                << "<svg " << m_width_height << " version=\"1.1\""
-                << std::endl
-                << "xmlns=\"http://www.w3.org/2000/svg\""
-                << std::endl
-                << "xmlns:xlink=\"http://www.w3.org/1999/xlink\""
-                << ">"
-                << std::endl;
         }
     }
 
+    void write_header(std::string const& width_height)
+    {
+        m_stream << "<?xml version=\"1.0\" standalone=\"no\"?>"
+            << std::endl
+            << "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\""
+            << std::endl
+            << "\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">"
+            << std::endl
+            << "<svg " << width_height << " version=\"1.1\""
+            << std::endl
+            << "xmlns=\"http://www.w3.org/2000/svg\""
+            << std::endl
+            << "xmlns:xlink=\"http://www.w3.org/1999/xlink\""
+            << ">"
+            << std::endl;
+    }
+
 public :
     
     /*!
-    \brief Constructor, initializing the SVG map. Opens and initializes the SVG. 
+    \brief Constructor, initializing the SVG map. Opens and initializes the SVG.
+         Should be called explicitly.
+    \param stream Output stream, should be a stream already open
+    \param width Width of the SVG map (in SVG pixels)
+    \param height Height of the SVG map (in SVG pixels)
+    \param scale Scale factor of the automatically determined bounding box.
+            1.0 fits perfectly. If it's not specified (other variant of the constructor),
+            the default is 0.95 to have a convenient margin around the geometries.
+    \param width_height Optional information to increase width and/or height
+    */
+    svg_mapper(std::ostream& stream
+        , SvgCoordinateType width
+        , SvgCoordinateType height
+        , calculation_type scale
+        , std::string const& width_height = "width=\"100%\" height=\"100%\"")
+        : m_stream(stream)
+        , m_width(width)
+        , m_height(height)
+        , m_scale(scale)
+    {
+        assign_inverse(m_bounding_box);
+        write_header(width_height);
```

Comment:
The header is now written immediately. Therefore uses can stream, for example arrow-definitions, right after the constructor (in previous version it needed to be done after `scale`).

Also the scale is here now and therefore I had to add a second version of the constructor (because having it after the width_height is inconvenient)

Reply:
Yes, the default `scale = 1` seems to be more intuitive.

Reply:
I'll change it

Reply:
:heavy_check_mark: changed to `1.0`